x-pack/plugins/apm/public/components/app/settings/storage_explorer/services_table/index.tsx

```diff
@@ -0,0 +1,295 @@
+          },
+        },
+      });
+    },
+    [indexLifecyclePhase, start, end, environment, kuery]
+  );
+
+  useEffect(() => {
+    // Closes any open rows when fetching new items
+    setItemIdToExpandedRowMap({});
+  }, [status]);
+
+  const loading =
+    status === FETCH_STATUS.NOT_INITIATED || status === FETCH_STATUS.LOADING;
+
+  const columns: Array<EuiBasicTableColumn<ValuesType<StorageExplorerItems>>> =
+    [
+      {
+        field: 'serviceName',
+        name: i18n.translate(
+          'xpack.apm.settings.storageExplorer.table.serviceColumnName',
+          {
+            defaultMessage: 'Service',
+          }
+        ),
+        sortable: true,
+        render: (_, { serviceName, agentName }) => {
+          const serviceLinkQuery = {
+            comparisonEnabled,
+            environment,
+            kuery,
+            rangeFrom,
+            rangeTo,
+            serviceGroup: '',
+          };
+
+          return (
+            <TruncateWithTooltip
+              data-test-subj="apmStorageExplorerServiceLink"
+              text={serviceName || NOT_AVAILABLE_LABEL}
+              content={
+                <ServiceLink
+                  query={serviceLinkQuery}
+                  serviceName={serviceName}
+                  agentName={agentName}
+                />
+              }
+            />
+          );
+        },
+      },
+      {
+        field: 'environment',
+        name: i18n.translate(
+          'xpack.apm.settings.storageExplorer.table.environmentColumnName',
+          {
+            defaultMessage: 'Environment',
+          }
+        ),
+        render: (_, { environments }) => (
+          <EnvironmentBadge environments={environments ?? []} />
+        ),
+        sortable: true,
+      },
+
+      {
+        field: 'sampling',
+        name: (
+          <EuiToolTip
+            content={i18n.translate(
+              'xpack.apm.settings.storageExplorer.table.samplingColumnDescription',
+              {
+                defaultMessage:
+                  'The ratio of sampled transactions to the total number of transactions.',
+              }
+            )}
+          >
+            <>
+              {i18n.translate(
+                'xpack.apm.settings.storageExplorer.table.samplingColumnName',
+                {
+                  defaultMessage: 'Sampling',
+                }
+              )}{' '}
+              <EuiIcon
+                size="s"
+                color="subdued"
+                type="questionInCircle"
+                className="eui-alignTop"
+              />
+            </>
+          </EuiToolTip>
+        ),
+        render: (value: string) => asPercent(parseFloat(value), 1),
+        sortable: true,
+      },
+      {
+        field: 'size',
+        name: (
+          <EuiToolTip
+            content={i18n.translate(
+              'xpack.apm.settings.storageExplorer.table.sizeColumnDescription',
+              {
+                defaultMessage:
```

Comment:
Could use some help here @bmorelli25

Reply:
Can you give me a bit more context into how this is calculated? I'm not sure I understand.

Reply:
Okay, looking at this again. It's the estimated disk size per service. Which is calculated by getting the number of documents for the service divided by the total number of documents. Then that number is multiplied by the total size of the cluster.

This is tough. Here are a few ideas that may or may not help you.

```
The estimated storage size of primary and replica shards per service. This estimate is calculated as a proportion of the total size of your indices based on the percentage of total documents that the service contains.
```

```
The estimated storage size per service. This estimate includes primary and replica shards and is calculated by prorating the total size of your indices by the service's document count divided by the total number of documents.
```

```
The estimated storage size per service. This estimate includes primary and replica shards and is calculated by comparing a service's document count to the total number of documents in all indices.
```

Reply:
That's great many thanks for the suggestions!